api_v3/services/LiveStreamService.php

```diff
@@ -577,4 +577,102 @@ public function regenerateStreamTokenAction($entryId)
 
 	}
 
+    /**
+     * Archive a live entry which was recorded
+     *
+     * @action archive
+     * @param string $liveEntryId
+     * @param KalturaLiveEntryArchiveOptions $archiveOptions
+     * @return bool
+     * @throws KalturaAPIException
+     * @throws KalturaClientException
+     * @throws PropelException
+     */
+	public function archiveAction($liveEntryId, KalturaLiveEntryArchiveOptions $archiveOptions)
+    {
+        $liveEntry = entryPeer::retrieveByPK($liveEntryId);
+        /** @var LiveStreamEntry $liveEntry */
+        if (!$liveEntry)
+        {
+            KalturaLog::err("Live entry id [" . $liveEntryId . "] not found");
+            throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $liveEntryId);
+        }
+        $cuePoints = CuePointPeer::retrieveByEntryId($liveEntryId);
+
+        $shouldBeDeletedCuePoints = array();
+
+        $notDeletedCuePointTagsArray = explode(",", $archiveOptions->notDeletedCuePointTags);
+        foreach ($cuePoints as $cuePoint)
+        {
+            /** @var CuePoint $cuePoint */
+            KalturaLog::info("Iterating over cue point [" . $cuePoint->getId() . "]");
+            $cuePointTags = explode(',', $cuePoint->getTags());
+            $containsImportantTag = false;
+            foreach ($notDeletedCuePointTagsArray as $importantCuePointTag)
```

Comment:
An optimization is required here.
1. If no tag was set on cue point iteration is not needed.
2. Since Since it we will less likely to have all on tags on a cuepointm than it is better to loop over the cuepoint tags, than to look each time on the list of tags.